Copy-pasting result computed by the calculator gives "Invalid input"

Copy-pasting result computed by the calculator (5.562684646268003457725581793331e-309) gives "Invalid input":
![image](https://user-images.githubusercontent.com/46918093/95691897-6c989c80-0c2b-11eb-829a-daccc3d546f0.png)

See also: https://github.com/microsoft/calculator/issues/1301.

Hello, can I take both issue #1301 and this issue?

I am also having some trouble building and running the program with visual studio. Namely, I got the following 2 build errors: 
1) 
4>C:\Users\User\Desktop\calculator\src\GraphControl\Generated Files\XamlTypeInfo.g.cpp(9,10): fatal error C1083: Cannot open precompiled header file: 'ARM\Debug\pch.pch.codeanalysis': No such file or directory
4>Done building project "GraphControl.vcxproj" -- FAILED.
2)  
8>C:\Users\User\Desktop\calculator\src\Calculator\Generated Files\XamlTypeInfo.g.cpp(9,10): fatal error C1083: Cannot open precompiled header file: 'ARM\Debug\Calculator.pch.codeanalysis': No such file or directory 

Would appreciate it if someone from the community can help!

--- a/src/CalcViewModel/Common/CopyPasteManager.cpp
+++ b/src/CalcViewModel/Common/CopyPasteManager.cpp
@@ -502,23 +502,32 @@ ULONG32 CopyPasteManager::StandardScientificOperandLength(Platform::String ^ ope
 {
     auto operandWstring = wstring(operand->Data());
     const bool hasDecimal = operandWstring.find('.') != wstring::npos;
+    auto length = operandWstring.length();
 
     if (hasDecimal)
     {
-        if (operandWstring.length() >= 2)
+        if (length >= 2)
         {
             if ((operandWstring[0] == L'0') && (operandWstring[1] == L'.'))
             {
-                return static_cast<ULONG32>(operandWstring.length() - 2);
+                length -= 2;
             }
             else
             {
-                return static_cast<ULONG32>(operandWstring.length() - 1);
+                length -= 1;
             }
         }
     }
 
-    return static_cast<ULONG32>(operandWstring.length());
+    auto exponentPos = operandWstring.find('e');
+    const bool hasExponent = exponentPos != wstring::npos;
+    if (hasExponent)
+    {
+        auto expLength = operandWstring.substr(exponentPos).length();
+        length -= expLength;
+    }
+
+    return static_cast<ULONG32>(length);
 }
 
 ULONG32 CopyPasteManager::ProgrammerOperandLength(Platform::String ^ operand, NumberBase numberBase)

--- a/src/CalculatorUnitTests/CopyPasteManagerTest.cpp
+++ b/src/CalculatorUnitTests/CopyPasteManagerTest.cpp
@@ -825,9 +825,13 @@ namespace CalculatorUnitTests
                                      L"12^2",
                                      L"-12.12^-2",
                                      L"61%99"
-                                     L"-6.1%99" };
+                                     L"-6.1%99",
+                                     L"1.1111111111111111111111111111111e+1142" };
         String
             ^ negativeInput[] = { L"abcdef", L"xyz",   L"ABab", L"e+234", L"123456789123456781234567890123456" /*boundary condition: greater than 32 digits*/,
+                                  L"11.1111111111111111111111111111111e+1142",
+                                  L"1.1e+10001", /*boundary condition: exponent greater than 5 digits*/
+                                  L"0.11111111111111111111111111111111111e+111111" /*boundary condition: both exponent and non exponent exceed limits*/
                                   L"SIN(2)", L"2+2==", L"2=+2" };
 
         ASSERT_POSITIVE_TESTCASES(ValidateScientificPasteExpression, positiveInput);

